udf-compiler/src/main/scala/com/nvidia/spark/udf/Instruction.scala

```diff
@@ -177,6 +179,7 @@ case class Instruction(opcode: Int, operand: Int, instructionStr: String) extend
       case Opcode.D2I | Opcode.F2I | Opcode.L2I => cast(state, IntegerType)
       case Opcode.D2L | Opcode.F2L | Opcode.I2L => cast(state, LongType)
       case Opcode.I2S => cast(state, ShortType)
+      case Opcode.CHECKCAST => checkcast(lambdaReflection, state)
```

Comment:
to the untrained eye, I am not sure why the `CHECKCAST` bytecode came up. Might be nice to add a comment for a reason for this one, just for future reference.

Reply:
you want us to explain why Scala compiler generates `checkcast`?

Reply:
Not exactly. We add feature X, and the PR says we are adding support for `() => Array.empty[T]` yet that carries a number of other things with it, just looking to understand what are the other things, at least on the surface.

Reply:
`checkcast` is from such expressions.

the bytecode sequence for `val f = () => Array.empty[Int]` is the following

```
         0: getstatic     #52                 // Field scala/Array$.MODULE$:Lscala/Array$;
         3: getstatic     #57                 // Field scala/reflect/ClassTag$.MODULE$:Lscala/reflect/ClassTag$;
         6: invokevirtual #61                 // Method scala/reflect/ClassTag$.Int:()Lscala/reflect/ClassTag;
         9: invokevirtual #65                 // Method scala/Array$.empty:(Lscala/reflect/ClassTag;)Ljava/lang/Object;
        12: checkcast     #67                 // class "[I"                                                                                                           15: areturn
```

I am not adding a number of other things.  I don't know why you thought that way.

Reply:
This helps @seanprime7.

> to the untrained eye, I am not sure why the CHECKCAST bytecode came up. Might be nice to add a comment for a reason for this one, just for future reference.

Your last message addresses this.